edkrepo/commands/sync_command.py

```diff
@@ -167,13 +171,20 @@ def run_command(self, args, config):
         manifest_repo = manifest.general_config.source_manifest_repo
         global_manifest_path = get_manifest_repo_path(manifest_repo, config)
         for repo_to_sync in repo_sources_to_sync:
+            if not args.fetch:
+                ui_functions.print_info_msg(humble.SYNCING.format(repo_to_sync.root, repo.active_branch), header = False)
```

Comment:
If you change 'repo.active_branch' to "repo_to_sync.branch" on line 177 and 179, it won't fail for the case where the repo is on a tag or commit instead of a branch.  The info message will say 'None branch' though, followed by the existing warning message that the repo is not on a branch.

    Syncing Edk2Platforms to latest None branch...
    No need to sync repo Edk2Platforms since it is in detached HEAD state